api/src/main/java/jakarta/data/metamodel/NumericExpression.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2025 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.metamodel;
+
+public interface NumericExpression<T, N extends Number & Comparable<N>>
+        extends ComparableExpression<T, N> {
+    NumericExpression<T,N> abs();
+    NumericExpression<T,N> negated();
+
+    NumericExpression<T,N> plus(N other);
+    NumericExpression<T,N> minus(N other);
+    NumericExpression<T,N> times(N other);
+    NumericExpression<T,N> divide(N other);
+
+    NumericExpression<T,N> plus(Expression<T,N> other);
+    NumericExpression<T,N> minus(Expression<T,N> other);
+    NumericExpression<T,N> times(Expression<T,N> other);
+    NumericExpression<T,N> divide(Expression<T,N> other);
```

Comment:
```suggestion
    NumericExpression<T,N> divideBy(N other);

    NumericExpression<T,N> plus(Expression<T,N> other);
    NumericExpression<T,N> minus(Expression<T,N> other);
    NumericExpression<T,N> times(Expression<T,N> other);
    NumericExpression<T,N> divideBy(Expression<T,N> other);
```
I like that `divide` is shorter, but it might read better to have `divideBy`.  I'm not really sure which is better, but I thought I'd bring it up.

An example,

```
        Restriction<Book> shortChaptered =
                _Book.numPages.divide(_Book.numChapters.plus(1))
                        .lessThanEqual(5);
```

or

```
        Restriction<Book> shortChaptered =
                _Book.numPages.divideBy(_Book.numChapters.plus(1))
                        .lessThanEqual(5);
```

Reply:
I'm not at all attached to the naming of these methods.